docs/core/testing/unit-testing-best-practices.md

```diff
@@ -0,0 +1,361 @@
+
+Writing tests for your code will naturally decouple your code, because it would be more difficult to test otherwise.
+
+## Characteristics of a good unit test
+- **Fast**. It is not uncommon for mature projects to have thousands of unit tests. Unit tests should take very little time to run. Milliseconds.
+- **Isolated**. Unit tests are standalone, can be run in isolation, and have no dependencies on any outside factors such as a file system or database.
+- **Repeatable**. Running a unit test should be consistent with its results, that is, it always returns the same result if you do not change anything in between runs.
+- **Self-Checking**. The test should be able to automatically detect if it passed or failed without any human interaction.
+- **Timely**. A unit test should not take a disproportionally long time to write compared to the code being tested. If you find testing the code taking a large amount of time compared to writing the code, consider a design that is more testable.
+
+## Let's speak the same language
+The term *mock* is unfortunately very misused when talking about testing. The following defines the most common types of *fakes* when writing unit tests:
+
+*Fake* - A fake is a generic term which can be used to describe either a stub or a mock object. Whether it is a stub or a mock depends on the context in which it's used. So in other words, a fake can be a stub or a mock.
+
+*Mock* - A mock object is a fake object in the system that decides whether or not a unit test has passed or failed. A mock starts out as a Fake until it is asserted against.
+
+*Stub* - A stub is a controllable replacement for an existing dependency (or collaborator) in the system. By using a stub, you can test your code without dealing with the dependency directly. By default, a fake starts out as a stub.
+
+Consider the following code snippet:
+
+```csharp
+var mockOrder = new MockOrder();
+var purchase = new Purchase(mockOrder);
+
+purchase.ValidateOrders();
+
+Assert.True(purchase.CanBeShipped);
+```
+
+This would be an example of stub being referred to as a mock. In this case, it is a stub. We're just passing in the Order as a means to be able to instantiate `Purchase` (the system under test). The name `MockOrder` is also very misleading because again, the order is not a mock.
+
+A better approach would be
+
+```csharp
+var stubOrder = new FakeOrder();
+var purchase = new Purchase(stubOrder);
+
+purchase.ValidateOrders();
+
+Assert.True(purchase.CanBeShipped);
+```
+
+By renaming the class to `FakeOrder`, we've made the class a lot more generic, the class can be used as a mock or a stub. Whichever is better for the test case. In the above example, `FakeOrder` is used as a stub. We're not using the `FakeOrder` in any shape or form during the assert. We just passed it into the `Purchase` class to satisfy the requirements of the constructor.
```

Comment:
This is the last change request (I promise 🤞 )

Starting with this paragraph, you switch from second person (you) to first person (we).  It should be 2nd person throughout.

Reply:
No worries, I appreciate the feedback! Just for clarity before I make too many edits, throughout the *entire* document you're looking for a 2nd person perspective? (i.e. replace we, we're, we've etc for you, you're, you've etc)

Looks like we've/we're was even used in the previous paragraph as well if that's what we're looking to do.

As an aside, it's funny you mention this because only a few days ago I asked a group of friends a question about this very same style.

![image](https://user-images.githubusercontent.com/6980197/43409640-8157a03e-93f2-11e8-998c-d78a171d8351.png)

Reply:
Thanks @jpreese And yes, the entire article should be 2nd person (I missed the earlier occurrence).